CONTRIBUTING.md

```diff
@@ -0,0 +1,41 @@
+# Coding guidelines
```

Comment:
I don't have any ideas for general guidelines yet, but I did come up with some coding guidelines and dumped them in for a start. Happy to take general guideline suggestions into the file as well - while keeping the obvious stuff out to avoid losing our reader.